pkg/proxy/apis/config/validation/validation.go

```diff
@@ -69,16 +76,23 @@ func Validate(config *kubeproxyconfig.KubeProxyConfiguration) field.ErrorList {
 	allErrs = append(allErrs, validateHostPort(config.MetricsBindAddress, newPath.Child("MetricsBindAddress"))...)
 
 	if config.ClusterCIDR != "" {
-		if config.FeatureGates[string(kubefeatures.IPv6DualStack)] {
-			cidrs := strings.Split(config.ClusterCIDR, ",")
-			for _, cidr := range cidrs {
-				if _, _, err := net.ParseCIDR(cidr); err != nil {
-					allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), cidr, "must be a valid CIDR block (e.g. 10.100.0.0/16 or FD02::0:0:0/96)"))
-				}
+		cidrs := strings.Split(config.ClusterCIDR, ",")
+		dualStackEnabled := effectiveFeatures.Enabled(kubefeatures.IPv6DualStack)
+
+		switch {
+		// if DualStack and more than one cidr validate if there is at least one of each IP family
+		case dualStackEnabled && len(cidrs) > 1:
+			isDual, err := netutils.IsDualStackCIDRStrings(cidrs)
```

Comment:
IsDualStackCIDRStrings allows more than 2 cidrs (e.g. `1.1.1.1,2.2.2.2,fde4:8dba:82e1::/48` would return `true, nil`). I expected us to limit the cidr list to 2 here

Reply:
it's limited to 2 cidrs when the `Proxier` object is instantiated here

https://github.com/kubernetes/kubernetes/blob/dcd0755f84bf8c15fd21e8d697b924991f641ae4/cmd/kube-proxy/app/server_others.go#L179-L181

 `cidrTuple` and `nodeTuple` return only a IPv4, IPv6 tuple
https://github.com/kubernetes/kubernetes/blob/dcd0755f84bf8c15fd21e8d697b924991f641ae4/cmd/kube-proxy/app/server_others.go#L320-L323

should we enforce it in validation too? 
(I'd like to come up with an standard way of validating dual stack and apply it to all the components, right now there is no consistency)

Reply:
> should we enforce it in validation too?

If that is an invalid configuration, yes. Whatever we can validate statically in configuration, we should.

Reply:
just to be clear, you mean erroring out or filtering the list?

Reply:
erroring